base/metadata/metadata.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2024, Cogent Core. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package metadata
+
+import (
+	"fmt"
+	"maps"
+)
+
+// Data is metadata as a map of named any elements
+// with generic support for type-safe Get and nil-safe Set.
+type Data map[string]any
+
+func (md *Data) init() {
+	if *md == nil {
+		*md = make(map[string]any)
+	}
+}
+
+// Set sets key to given value, ensuring that
+// the map is created if not previously.
+func (md *Data) Set(key string, value any) {
+	md.init()
+	(*md)[key] = value
+}
+
+// Get gets metadata value of given type.
+// returns error if not present or item is a different type.
+func Get[T any](md Data, key string) (T, error) {
```

Comment:
Do you really need this error?

Reply:
yes.  if you have inconsistency in typing, you want to be able to know and fix that vs. just randomly failing.